### ISSUE TITLE
For drmm model, can I use other value as label?

### ISSUE THREAD

**Issue by czt616** [NTMC-Community/MatchZoo-py#70]
I am wondering if I can use label = 0.5 or other float value as label to train and test drmm model? How about duet and matchpyramid?

**Comment by Chriskuei** [comment]
Whether it can be used depends on the task  and the number of labels.

#### Two labels
Map labels to `[0, 1]`, then you can use all models in MatchZoo with any loss

#### Three or more labels
- Classification task: Map labels to index (i.g. `[0, 1, 2]` for three labels)
- Ranking task
  - You can use `MSELoss` by point-wise training without any changes
  - For losses like `RankHingeLoss` and `RankCrossEntropyLoss`, current version only supports for binary labels

**Comment by czt616** [comment]
I have some more questions.

1) I added some new documents with a label of 0.75 to wikiqa train set. However, DRMM model still could be trained. Why is this happening?

2) In the /tutorials/ranking/init.ipynb. It use:
ranking_task = mz.tasks.Ranking(losses=mz.losses.RankHingeLoss())
Then in the drmm.ipynb, it use:
ranking_task = mz.tasks.Ranking(losses=mz.losses.RankCrossEntropyLoss(num_neg=10))

So, which loss function does it finally use？If I want to use MSE loss. How should I change it?

3) In the /matchzoo/losses/rank_cross_entropy_loss.py
The return value for forward function: return -torch.mean(torch.sum(labels * torch.log(F.softmax(logits, dim=-1)), dim=-1)
Is labels are some thing like 0,0,1,0,1?

**Comment by Chriskuei** [comment]
1. For `RankHingeLoss`, it will learn a margin in a pair during training. If you have labels with 0, 0.75, 1, pairs with different margins could be still trained, but the different margins can not be learned (`RankHingeLoss` only supports one margin during training).

For `RankCrossEntropyLoss`, https://github.com/NTMC-Community/MatchZoo-py/blob/8883d0933a62610d71fec0215dce643630e03b1c/matchzoo/losses/rank_cross_entropy_loss.py#L36-L38
Since multiplied by `labels`, only when the labels are binary labels, it denotes the probability of each label.

2. It finally use `ranking_task = mz.tasks.Ranking(losses=mz.losses.RankCrossEntropyLoss(num_neg=10))`. If you want to use MSE loss, you need to define the task as `ranking_task = mz.tasks.Ranking(losses='mse')`.

3. Yes, see what I explained above.
